tracer/test/Datadog.Trace.Tests/ContinuousProfiler/ContextTrackerTests.cs

```diff
@@ -0,0 +1,71 @@
+// <copyright file="ContextTrackerTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.ContinuousProfiler;
+using Datadog.Trace.Sampling;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace Datadog.Trace.Tests.ContinuousProfiler
+{
+    [Collection(nameof(ContextTrackerTests))]
+    [CollectionDefinition(nameof(ContextTrackerTests), DisableParallelization = true)]
+    public class ContextTrackerTests
+    {
+        [Fact]
+        public void SetEndpointOnRootWebSpans()
+        {
+            const string expectedEndpoint = "My endpoint";
+
+            var contextTracker = new Mock<IContextTracker>();
+
+            var invocations = new List<(ulong SpanId, string Endpoint)>();
+
+            contextTracker.Setup(c => c.SetEndpoint(It.IsAny<ulong>(), It.IsAny<string>()))
+                          .Callback<ulong, string>((i, e) => invocations.Add((i, e)));
+
+            try
+            {
+                Profiler.SetInstanceForTests(new Profiler(contextTracker.Object, null));
+
+                var tracer = CreateTracer();
+
+                var rootWebScope = tracer.StartActive("Root");
```

Comment:
nit: To make the scope usage similar to customer code, you could wrap the scopes in `using` statements. At least for now we expect them to be disposable so this could make sure the end-to-end disposable use cases work correctly

Reply:
Changed in https://github.com/DataDog/dd-trace-dotnet/pull/3015/commits/520872d8381d8947b26576f1ca8e25441d3627d7

It actually helped me spot an error in the test 😅